docs/app/pages/components/sections/utilities/color-service/snippets/app.ts

```diff
@@ -0,0 +1,165 @@
+import { BehaviorSubject } from 'rxjs/BehaviorSubject';
+import { ColorService, ThemeColor } from 'ux-aspects';
+import { Component, OnInit } from '@angular/core';
+
+@Component({
+    selector: 'app',
+    templateUrl: './src/app.component.html',
+    styleUrls: ['./src/app.component.css']
+})
+export class AppComponent implements OnInit {
+
+    // ux-select configuration properties
+    maxHeight: string = '250px';
+ 
+    // Customize settings
+    colorSet = new BehaviorSubject<string>('colors');
+    
+    private colorSets: { sales?: any[] } = {};
+
+    selectedColorSet(): any[] {
+        return this.colorSets[this.colorSet.getValue()];
+    }
+
+    // configure the directive data
+    donutChartData: Chart.ChartData = [{
+        data: [33, 34, 33],
+        borderWidth: 0
+    }];
+
+    donutChartOptions: Chart.ChartOptions;
+    donutChartColors: any;
+
+    constructor(public colorService: ColorService) {
+
+        // color list
+        this.colorSets.colors = ['Primary', 'Accent', 'Secondary', 'Alternate1', 'Alternate2', 'Alternate3', 'Vibrant1', 'Vibrant2',
+                'Grey1', 'Grey2', 'Grey3', 'Grey4', 'Grey5', 'Grey6', 'Grey7', 'Grey8', 'Chart1', 'Chart2', 'Chart3', 'Chart4', 'Chart5',
+                'Chart6', 'OK', 'Warning', 'Critical'];
+
+        let tooltipBackgroundColor = colorService.getColor('grey2').toHex();
+
+        this.donutChartOptions = {
+            maintainAspectRatio: false,
+            responsive: true,
+            cutoutPercentage: 70,
+            tooltips: {
+                callbacks: {
+                    title: (item: Chart.ChartTooltipItem[]) => {
+                        return;
+                    },
+                    label: (item: Chart.ChartTooltipItem, data: any) => {
+
+                        // get the dataset
+                        let dataset = data.datasets[0];
+
+                        // calculate the total of all segment values
+                        let total = dataset.data.reduce((previousValue: any, currentValue: any) => {
+                            return previousValue + currentValue;
+                        });
+
+                        // get the value of the current segment
+                        let segmentValue = dataset.data[item.index];
+
+                        // calculate the percentage of the current segment compared to the total
+                        let precentage = Math.round(((segmentValue / total) * 100));
+
+                        return `${ precentage }%, Sales ${ item.index + 1 }`;
```

Comment:
Let's cut some of this out, since we don't need to demo donut chart features.

````javascript
label: (item: Chart.ChartTooltipItem, data: any) => {
  return `Sales ${ item.index + 1 }`;
}
````